src/scripts/install_machine.py

```diff
@@ -0,0 +1,362 @@
+#! /usr/bin/python3
+
+import traceback
+import json
+import subprocess
+import sys
+import os
+import xml.etree.ElementTree as ET
+import tempfile
+import logging
+
+
+def virsh(connectionName, args):
+    cmd = ["virsh", "-c", f"qemu:///{connectionName}"]
+    for arg in args:
+        cmd.append(arg)
+
+    logging.debug("Running virsh command :" + ' '.join(cmd))
+
+    return subprocess.run(cmd, check=False, capture_output=True)
```

Comment:
Here's a bit more Pythonic and more convenient to use (you'll need to update all call sites to remove the array around the paramters, and just pass them as directly to parameters of the function), and also just read stdout directly.

```suggestion
def virsh(connection, *args, check=True):
    cmd = ("virsh", "-c", f"qemu:///{connection}") + args
    logging.debug("Running virsh command :" + ' '.join(cmd))
    return subprocess.run(cmd, check=check, capture_output=True).stdout
```

or just use `check_output()` and catch expected exceptions.